src/test/java/org/mule/test/http/api/HttpResponseAttributesTestCase.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.test.http.api;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+
+import org.mule.extension.http.api.HttpResponseAttributes;
+import org.mule.runtime.api.util.MultiMap;
+
+import org.junit.Test;
+
+public class HttpResponseAttributesTestCase {
+
+  private static final String COMPLETE_TO_STRING = "org.mule.extension.http.api.HttpResponseAttributes\n" +
+      "{\n" +
+      "   Status Code=401\n" +
+      "   Reason Phrase=Some Reason Phrase\n" +
+      "   Headers=[\n" +
+      "      header2=value2\n" +
+      "      header1=value1\n" +
+      "   ]\n" +
+      "}";
+
+  private static final String TO_STRING_WITHOUT_HEADERS = "org.mule.extension.http.api.HttpResponseAttributes\n" +
+      "{\n" +
+      "   Status Code=401\n" +
+      "   Reason Phrase=Some Reason Phrase\n" +
+      "   Headers=[\n" +
+      "   ]\n" +
```

Comment:
I think that if you add the line separator before new lines instead of at the end you can get `\n Headers=[]` for the empty case.